src/main/java/com/example/rentabookrestservices/repository/SaleRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.example.rentabookrestservices.repository;
+
+import com.example.rentabookrestservices.domain.Sale;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import javax.transaction.Transactional;
+
+public interface SaleRepository extends JpaRepository<Sale, Long> {
+    Sale findByOperationNumber(String operationNumber);
+    boolean existsByOperationNumber(String operationNumber);
+
+    @Transactional
```

Comment:
Bu anatasyona neden ihtiyaç duydun @abmehmet açıklayabilir misin ?

Reply:
abi, kiralama işlemini iptal ederken silme işlemi yapıyorum, bu yüzden bu anotasyon sayesinde veri tabanından silme işlemini sağlıyrum, arka planda şöyle gerçekleşiyor abi, silme işlemi için fiş numarasını alıyorum, var mı diye sorguuyorum, bu sorgu sonucunda eğer böyle bir kiralama var ise onu siliyoum, zannediyorum abi, sorgulama işlemi bitmeden silme işlemine geçilmesii önlemek için bu şekilde bir kullanımı var, o yüzden eklemiştim abi, bu anotasyonu kullanmadığım zaman silme işlemi gerçekleşmiyor abi